optaplanner-core/src/main/java/org/optaplanner/core/config/util/ConfigUtils.java

```diff
@@ -78,6 +78,18 @@
         return ObjectUtils.equals(a, b) ? a : null;
     }
 
+    public static <T> T meldProperty(T a, T b) {
```

Comment:
The difference between meld and merge isn't inherently clear. Please add a javadoc to explain the difference or rename both methods more self-explanatory.